docs/manual/key-storage/enterprise-runner-key-storage.md

```diff
@@ -0,0 +1,132 @@
+network as Vault and retrieve secrets from Vault to be used by operations on the Runner.
+
+![Key Storage Through Runner](@assets/img/key-storage-runner-browse.png)<br>
+
+:::warning Key Storage Plugin Support
+As of version **`4.16.0`**, integration with Hashicorp Vault is supported on the Runner.<br>
+In future releases, integration with **CyberArk** and **Thycotic** will also be available through the Runner.
+:::
+
+## Use Cases
+1. **Remote Node Commands & Scrips** (Node Executors & File Copiers): When SSH or WinRM credentials are stored in a secrets provider, then the Runner can retrieve keys from the provider to authenticate with remote nodes in order to execute commands or scripts.<br><br>
+2. **Internal Tools & Infrastructure** (Job step plugins): When executing Jobs that include steps that integrate with internal tools APIs - such as Jira, Jenkins, homegrown tooling, etc. - or infrastructure such as databases, then the Runner can use secrets to authenticate with these endpoints using best-practice security standards.<br><br>
+3. **Inventory Discovery** (Node Sources): The Runner can be used to discover inventory in secure or remote environments. By retrieving keys from a secrets-provider, the Runner can authenticate with an API endpoint, such as the VMware vSphere API, in order to retrieve the inventory.
+
+:::warning Support for Node Discovery and Remote Commands
+As of version **`4.16.0`**, secrets from the Runner can only be used for Job step plugins - such as HTTP, SQL, Kubernetes, and so on.<br>
+In upcoming releases, support for remote-node commands and scripts as well as inventory discovery will be supported as well.
+:::
+
+## Handling Secrets
+Given the sensitivity of secrets retrieved by the Runner, the following guardrails have been put in place:
+
+1. Secrets retrieved by a Runner can only be used for operations executed on that Runner.  These secrets are not sent back to the Process Automation cluster or Runbook Automation instance.  Therefore, these secrets can not be used for plugins that do not reside on the Runner.<br><br>
+2. Secrets are masked in logs, including Job execution logs.  Any secrets retrieved by the Runner and printed to the Job log output will appear as `[SECURE]` in the logs.<br><br>
+3. Sensitive environment variables set on the Runner are masked in logs, including Job execution logs.  Any environment variables ending with `TOKEN` or `KEY` or `PASSWORD` will be printed as `[SECURE]` in the logs.
+
+## Configuration
+To configure a Key Storage integration on the Runner, **configuration properties are set on the Runner**. These properties can be set through the following methods:
+
+**Note:** These examples do not include the full list of configuration properties to configure an integration with Vault.  The full list of
+configuration properties can be found [here](/manual/key-storage/storage-plugins/vault.html#configuration).
+
```

Comment:
For the three examples I think we should have a recommended approach between YAML and Env variable.

If they do YAML and don't include other settings the Runner needs to operate (url for example) will it still work?

Reply:
Why do we need to be opinionated about whether customers use YAML vs. Env variable?

And yes: the YAML can include just the key storage config properties.  the connection properties for communicating with the server are defined elsewhere.

Reply:
Given multiple options how do they decide which to use?

Reply:
resolved in DM